join.spec.md

```diff
@@ -413,23 +413,40 @@ The {@join__graph} directive MUST be applied to each enum value on {join__Graph}
 
 ##! @join__type
 
-Declares an entity key for a type on a subgraph.
+Declares that a type is known by a particular subgraph. Optionally, an entity key for the type on that subgraph may be specified.
 
 ```graphql definition
 directive @join__type(
   graph: join__Graph!
-  key: String!
+  key: String
 ) repeatable on OBJECT | INTERFACE
 ```
 
-When this directive is placed on a type `T`, it means that subgraph `graph` MUST be able to:
+When this directive is placed on a type `T`, it means that subgraph `graph` contains a definition for type `T`.
+
+If `key` is non-null, it means that subgraph `graph` MUST be able to:
 - Resolve selections on objects of the given type that contain the field set in `key`
 - Use `Query._entities` to resolve representations of objects containing `__typename: "T"` and the fields from the field set in `key`
 
 :::[example](photos.graphql#Image) -- Using {@join__type} to specify subgraph keys
 
 Every type with a {@join__type} MUST also have a [{@join__owner}](#@join__owner) directive. Any type with a [{@join__owner}](#@join__owner) directive MUST have at least one {@join__type} directive with the same `graph` as the [{@join__owner}](#@join__owner) directive (the "owning graph"), and MUST have at most one {@join__type} directive for each `graph` value other than the owning graph. Any value that appears as a `key` in a {@join__type} directive with a `graph` value other than the owning graph must also appear as a `key` in a {@join__type} directive with `graph` equal to the owning graph.
 
+##! @join__implements
+
+Declares that a type implements an interface in a particular subgraph.
+
+```graphql definition
+directive @join__implements(
```

Comment:
we could address https://github.com/apollographql/specs-join/issues/17 by expanding this to cover unions.

one option for doing that:

```graphql
directive @join__implements(
  graph: join__Graph!,
  interfaces: [String!],
  unions: [String!]
) repeatable on OBJECT | INTERFACE | UNION
```

Reply:
Ok with that, but (and it's probably a typo but to make sure we're on the same page): we don't want to allow the directive on `UNION`, right?  The directive is applied on the type that "implements" another, but union cannot be part of other unions, nor implement interfaces.